### PR TITLE
feat: optimize scheduler start server

### PR DESCRIPTION
**Pull request by gaius-qi**
Signed-off-by: Gaius <gaius.qi@gmail.com>

## Description
- optimize scheduler start server

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: scheduler/config/config.go [hunk 1]
```diff
@@ -39,19 +39,78 @@ type Config struct {
...
+			LocalWorkerNum:     10,
+			Redis: &RedisConfig{
+				Host:      "",
+				Port:      6379,
+				Password:  "",
+				BrokerDB:  1,
+				BackendDB: 2,
+			},
+		},
 	}
```
Comment (244372610) [inline scheduler/config/config.go:114]: What are the benefits of the configuration items taken together

Comment (gaius-qi) [reply scheduler/config/config.go:114]: Look at other module config.

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -144,28 +157,26 @@ func (s *Server) Serve() error {
...
 	logger.Infof("start server at port %d", port)
 	var opts []grpc.ServerOption
 	if s.config.Options.Telemetry.Jaeger != "" {
 		opts = append(opts, grpc.ChainUnaryInterceptor(otelgrpc.UnaryServerInterceptor()), grpc.ChainStreamInterceptor(otelgrpc.StreamServerInterceptor()))
 	}
 	if err := rpc.StartTCPServer(port, port, s.schedulerServer, opts...); err != nil {
-		return err
+		logger.Fatalf("grpc start failed %v", err)
 	}
+	defer s.Stop()
```
Comment (244372610) [inline scheduler/scheduler.go:169]: Does s.top() have to execute if something goes wrong with the previous code

File: scheduler/scheduler.go [hunk 3]
```diff
@@ -144,28 +157,26 @@ func (s *Server) Serve() error {
...
-		s.schedulerService.Stop()
-		s.job.Stop()
-	}
-	return
+func (s *Server) Stop() {
+	rpc.StopServer()
+	s.schedulerService.Stop()
+	s.dynConfig.Stop()
+	s.managerConn.Close()
+	s.job.Stop()
```
Comment (244372610) [inline scheduler/scheduler.go:179]: job is nil ?

File: scheduler/scheduler.go [hunk 4]
```diff
@@ -144,28 +157,26 @@ func (s *Server) Serve() error {
...
 	if s.config.Options.Telemetry.Jaeger != "" {
 		opts = append(opts, grpc.ChainUnaryInterceptor(otelgrpc.UnaryServerInterceptor()), grpc.ChainStreamInterceptor(otelgrpc.StreamServerInterceptor()))
 	}
 	if err := rpc.StartTCPServer(port, port, s.schedulerServer, opts...); err != nil {
-		return err
+		logger.Fatalf("grpc start failed %v", err)
 	}
+	defer s.Stop()
+
 	return nil
```
Comment (244372610) [inline scheduler/scheduler.go:171]: rpc err not be returned ?

**Review by jim3ma — APPROVED** [review]

**Review by yxxhero — APPROVED** [review]
LGTM

**Pull request merged**